tests/test_performance.py

```diff
@@ -0,0 +1,145 @@
+    Y_rv = SwitchingProcess(
+        "Y_t",
+        [pm.Constant.dist(0), pm.Constant.dist(mu)],
+        V_rv,
+        observed=observed,
+    )
+    return Y_rv
+
+
+def assert_posterior_predictives(model, Y, X, test_X, test_y, xi_true, starting_state):
+    number_of_draws = 100
+
+    with model:
+        steps = [
+            FFBSStep([model.V_t]),
+            pm.NUTS(
+                vars=[
+                    model.gamma_0,
+                    model.Gamma,
+                ],
+                target_accept=0.90,
+            ),
+        ]
+
+    with model:
+        posterior_trace = pm.sample(
+            draws=number_of_draws,
+            step=steps,
+            random_seed=100,
+            return_inferencedata=True,
+            chains=1,
+            cores=1,
+            tune=number_of_draws // 2,
+            progressbar=True,
+            idata_kwargs={"dims": {"Y_t": ["dt"]}},
+        )
+
+    # Update the shared variable values
+    Y.set_value(np.ones(test_X.shape[0]))
+    X.set_value(test_X)
+    posterior_trace.posterior.gamma_0.values[0] = np.array([starting_state])[
+        np.repeat(0, number_of_draws)
+    ]
+
+    model.V_t.distribution.shape = (test_X.shape[0],)
+
+    assert np.abs(posterior_trace.posterior.xis - xi_true).mean() < 1.5
+
+    trace = posterior_trace.posterior.drop_vars(["Gamma", "V_t"])
+
+    with theano.config.change_flags(compute_test_value="off"):
+        adds_pois_ppc = pm.sample_posterior_predictive(
+            trace, var_names=["V_t", "Y_t", "Gamma"], model=model
+        )
+
+    assert np.abs(adds_pois_ppc["V_t"] - (test_y > 0) * 1).mean() < 0.3
+
+
+@pytest.mark.slow
+def test_time_variant_model_base():
+    df = gen_toy_data()
+
+    formula_str = "1 + C(weekday)"
+    X_df = patsy.dmatrix(formula_str, df, return_type="dataframe").values
+
+    xi_shape = X_df.shape[1]
+
+    xi_0_true = np.array([2.0, -2.0, 2.0, -2.0, 2.0, -2.0, 2.0]).reshape(xi_shape, 1)
+    xi_1_true = np.array([2.0, -2.0, 2.0, -2.0, 2.0, -2.0, 2.0]).reshape(xi_shape, 1)
+
+    xis_rv_true = np.stack([xi_0_true, xi_1_true], axis=1)
+
+    with pm.Model(theano_config={"compute_test_value": "ignore"}) as sim_model:
+        _ = create_poisson_zero_hmm(
+            X_df, mu=1000, xis=xis_rv_true, observed=np.zeros(X_df.shape[0])
+        )
+
+    sim_point = pm.sample_prior_predictive(samples=1, model=sim_model)
+
+    y_t = sim_point["Y_t"].squeeze()
+
+    split = int(len(y_t) * 0.7)
+
+    train_y, test_y = y_t[:split], y_t[split:]
+    train_X, test_X = X_df[:split, :], X_df[split:, :]
+
+    X = shared(train_X, name="X", borrow=True)
+    Y = shared(train_y, name="y_t", borrow=True)
+
+    with pm.Model() as model:
+        xis_rv = pm.Normal("xis", 0, 10, shape=xis_rv_true.shape)
+
+        _ = create_poisson_zero_hmm(X, 1000, xis_rv, Y)
+
+    last_state = (train_y[-1] > 0) * 1
+    starting_state = [1 - last_state, last_state]
+
+    return assert_posterior_predictives(
```

Comment:
As always, unless this function is called at least more than once, there's no need to make it a separate function.  

If this function is a copy of an existing function, then we should import that function.  We can also modify existing functions so that they work for more than one situation.